src/fsharp/service/SemanticClassification.fs

```diff
@@ -0,0 +1,183 @@
+// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.SourceCodeServices
+
+[<RequireQualifiedAccess>]
+type SemanticClassificationType =
+    | ReferenceType
+    | ValueType
+    | UnionCase
+    | Function
+    | Property
+    | MutableVar
+    | Module
+    | Printf
+    | ComputationExpression
+    | IntrinsicFunction
+    | Enumeration
+    | Interface
+    | TypeArgument
+    | Operator
+    | Disposable
+
+[<AutoOpen>]
+module TcResolutionsExtensions =
+    open System.Diagnostics
+    open System.Collections.Generic
+    open System.Collections.Immutable
+
+    open FSharp.Core.Printf
+    open FSharp.Compiler.AbstractIL.Internal.Library  
+
+    open FSharp.Compiler
+    open FSharp.Compiler.Range
+    open FSharp.Compiler.Tast
+    open FSharp.Compiler.Infos
+    open FSharp.Compiler.NameResolution
+    open FSharp.Compiler.ErrorLogger
+    open FSharp.Compiler.Lib
+    open FSharp.Compiler.PrettyNaming
+    open FSharp.Compiler.Tastops
+    open FSharp.Compiler.TcGlobals 
+    open FSharp.Compiler.SourceCodeServices.SymbolHelpers 
+
+    let (|CNR|) (cnr:CapturedNameResolution) =
+        (cnr.Pos, cnr.Item, cnr.ItemOccurence, cnr.DisplayEnv, cnr.NameResolutionEnv, cnr.AccessorDomain, cnr.Range)
+
+    type TcResolutions with
+
+        member sResolutions.GetSemanticClassification(g: TcGlobals, amap: Import.ImportMap, formatSpecifierLocations: (range * int) [], range: range option) : struct(range * SemanticClassificationType) [] =
+              ErrorScope.Protect Range.range0 
+               (fun () -> 
+                let (|LegitTypeOccurence|_|) = function
+                    | ItemOccurence.UseInType
+                    | ItemOccurence.UseInAttribute
+                    | ItemOccurence.Use _
+                    | ItemOccurence.Binding _
+                    | ItemOccurence.Pattern _ -> Some()
+                    | _ -> None
+
+                let (|OptionalArgumentAttribute|_|) ttype =
+                    match ttype with
+                    | TType.TType_app(tref, _) when tref.Stamp = g.attrib_OptionalArgumentAttribute.TyconRef.Stamp -> Some()
+                    | _ -> None
+
+                let (|KeywordIntrinsicValue|_|) (vref: ValRef) =
+                    if valRefEq g g.raise_vref vref ||
+                       valRefEq g g.reraise_vref vref ||
+                       valRefEq g g.typeof_vref vref ||
+                       valRefEq g g.typedefof_vref vref ||
+                       valRefEq g g.sizeof_vref vref ||
+                       valRefEq g g.nameof_vref vref
+                    then Some()
```

Comment:
The compiler code pretty much always puts then at end of previous line

Reply:
sure

Reply:
Note, this was the previous code for semantic classification.